crates/cxx-qt-gen/src/parser/qobject.rs

```diff
@@ -126,7 +152,8 @@ impl ParsedQObject {
     }
 
     /// Extract all the properties from [syn::Fields] from a [syn::ItemStruct]
-    pub fn parse_struct_fields(&mut self, fields: &mut Fields) -> Result<()> {
+    pub fn parse_struct_fields(fields: &mut Fields) -> Result<Vec<ParsedQProperty>> {
```

Comment:
How about making this private and changing test_parse_struct_fields_valid to use the `properties` field of the QObject returned from `from_struct`?

Reply:
This can just be made private anyway and the tests could still reach it. But i think it makes sense to only unit test public methods anyway, so i'll change it to the from_struct...

Reply:
Done